File: Lezioni/10.ListView-con-Adapter/Lista_semplice_di_stringhe/app/src/main/java/com/example/lista_semplice_di_stringhe/MainActivity.java
package com.example.lista_semplice_di_stringhe;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;



public class MainActivity extends AppCompatActivity {
    private ListView listView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = findViewById(R.id.list_view);

        // Dati: semplice array di stringhe
        String[] frutti = {
                "Mela",
                "Banana",
                "Arancia",
                "Fragola",
                "Uva"
        };

        // ArrayAdapter con layout predefinito (simple_list_item_1)
        ArrayAdapter<String> adapter = new ArrayAdapter<>(
                this,
                android.R.layout.simple_list_item_1,
                frutti
        );

        // Assegna l'adapter alla ListView
        listView.setAdapter(adapter);

        // Click listener: reagire ai tap sugli elementi
        listView.setOnItemClickListener((parent, view, position, id) -> {
            String selected = frutti[position];
            Toast.makeText(MainActivity.this, "Selezionato: " + selected, Toast.LENGTH_SHORT).show();
        });
    }

}

File: Lezioni/01.Realizzazione-e-test-di-una-semplice-app/Esercizi/1.Contatore_digitale_con_reset/app/src/main/java/com/example/a1contatore_digitale_con_reset/MainActivity.kt
package com.example.a1contatore_digitale_con_reset

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    // Variabile di stato
    private var count = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Collega il file Kotlin al file XML
        setContentView(R.layout.activity_main)

        val textView = findViewById<TextView>(R.id.text_counter)
        val btnIncrement = findViewById<Button>(R.id.button_increment)
        val btnReset = findViewById<Button>(R.id.button_reset) // Riferimento al nuovo ID

        btnIncrement.setOnClickListener {
            count++
            textView.text = count.toString()
        }

        btnReset.setOnClickListener {
            count = 0 // Logica: azzeramento variabile
            textView.text = count.toString() // UI: aggiornamento manuale obbligatorio
        }
    }
}